### ISSUE TITLE
ModuleVersionDetails should have nativeBackends

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-module-resolver#129]
The `ModuleVersionDetails` class should have an attribute that tells me what are the supported `Backends`.

@quintesse would you please take a look at this. Thanks.

**Comment by quintesse** [comment]
I don't think I can do this before the deadline you set for a wednesday release. What's the specific reason you need this feature? I'd think that knowing there is an implementation (native or not) would be sufficient?

**Comment by quintesse** [comment]
Ah wait you linked another issue.

**Comment by gavinking** [comment]
I want to automatically add `native` annotations on the `import` statements. I already have a branch of the IDE which implements this, but I can't get the information I need from a module query.

**Comment by quintesse** [comment]
But couldn't you , for now, use the fact that a query for JVM + JS only returns one or the other for native modules?

**Comment by gavinking** [comment]
Well sure I guess in principle I could do it with two queries, but that would make something that is already too slow now twice as slow.

**Comment by quintesse** [comment]
Well I mean that if somebody is putting in an import statement we have to retrieve the artifacts anyway, right? So the moment you retrieve them you can check which ones you got. That should involve any queries AFAICT.

**Comment by gavinking** [comment]
No, because as-per the resolution to https://github.com/ceylon/ceylon-module-resolver/issues/128, it only sends me the artifacts I asked for. It doesn't tell me that the module _also_ has other artifacts available.

**Comment by quintesse** [comment]
But that's what I mean, you'd normally ask for *both* artifacts, right?
Because I see two situations:

 - you have a module marked `native("xxx")` so you only ask for artifact XXX and don't ahve to worry about putting `native` on any imports
 - the module is cross-platform so you ask for both JVM and JS and then look at which you got (you either get none, one or both)

THat should be enough information, right?

**Comment by gavinking** [comment]
> But that's what I mean, you'd normally ask for both artifacts, right?

No, I don't, because I want to get a filtered list of modules. Instead I ask for only the modules that are compatible with at least one of the enabled backends.

**Comment by gavinking** [comment]
@quintesse Note that I can have a `native` import in a cross-platform module.

**Comment by gavinking** [comment]
Reassigning to 1.3 because I think I can live without this for now.

**Comment by quintesse** [comment]
> Note that I can have a native import in a cross-platform module.

Yes, that was the second bullet point I mentioned above.

> No, I don't, because I want to get a filtered list of modules. Instead I ask for only the modules that are compatible with at least one of the enabled backends.

I'm not sure I understand. If you ask for the list of modules that are compatible with either JVM **or** JS then you have your answer right there, don't you? Some of the modules will have JVM artifacts, some JS and some both. It would seem to me that's enough information?
